src/main/java/com/splunk/logging/HttpEventCollectorSender.java

```diff
@@ -248,6 +295,32 @@ private String serializeEventInfo(HttpEventCollectorEventInfo eventInfo) {
         return event.toString();
     }
 
+    @SuppressWarnings("unchecked")
+    private JSONObject writeThrowableInfoToJson(HttpEventCollectorThrowableInfo throwableInfo) {
+        if (throwableInfo == null) {
+            return null;
+        }
+        
+        final JSONObject json = new JSONObject();
+        putIfPresent(json, "throwable_message", throwableInfo.getMessage());
+        putIfPresent(json, "throwable_class", throwableInfo.getClassName());
+        
+        final JSONArray stackTrace = new JSONArray();
+        json.put("stack_trace", stackTrace);
+        if (throwableInfo.getStackTraceElements() != null) {
```

Comment:
There is no strong guarantee that it will not be. It seems better to be cautiously defensive here.